### PR TITLE
Add Server::resetState method

### PR DESCRIPTION
**Pull request by laszlof**
Add functionality to reset the state of a VM.

### CODE REVIEW

File: src/Compute/v2/Models/Server.php [hunk 1]
```diff
@@ -183,6 +183,17 @@ public function changePassword(string $newPassword)
         ]);
     }
 
+    /**
+     * Issue a resetState call to the server.
+     */
+    public function resetState()
+    {
+        $response = $this->execute($this->api->resetServerState(), [
```
Comment (haphan) [inline src/Compute/v2/Models/Server.php]: Remove `$response`

File: src/Compute/v2/Params.php [hunk 2]
```diff
@@ -15,6 +15,19 @@ public function urlId(string $type): array
         ]);
     }
 
+    public function osResetState(): array
```
Comment (haphan) [inline src/Compute/v2/Params.php]: use `resetState`

File: src/Compute/v2/Api.php [hunk 3]
```diff
@@ -279,6 +279,19 @@ public function changeServerPassword(): array
...
 
+    
+    public function resetServerState(): array
+    {
+        return [
+            'method' => 'POST',
+            'path'   => 'servers/{id}/action',
+            'params' => [
+                'id'         => $this->params->urlId('server'),
+                'resetState' => $this->params->osResetState()
```
Comment (haphan) [inline src/Compute/v2/Api.php]: use `resetState`

**Comment by haphan** [comment]
@laszlof I left some minor change requests :)

**Comment by laszlof** [comment]
Yup, saw. Was away for the weekend, will take a look today.

On Feb 5, 2018 12:18 AM, "Ha Phan" <notifications@github.com> wrote:

@laszlof <https://github.com/laszlof> I left some minor change requests :)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/php-opencloud/openstack/pull/196#issuecomment-362984236>,
or mute the thread

**Comment by laszlof** [comment]
@haphan The issues have been corrected.

**Pull request merged**